homework/insensitive-palindrom/palindrome.cpp

```diff
@@ -0,0 +1,20 @@
+#include <algorithm>
+#include <cctype>
+#include <iterator>
+#include <iostream>
+
+#include "palindrome.hpp"
+
+bool is_palindrome(const std::string& str) {
+
+	std::string cleaned_str{};
+	std::copy_if(str.begin(), str.end(), 
+		     std::back_inserter(cleaned_str),
+		     [](auto c) { return isalnum(c); });
```

Comment:
or `remove_copy_if` and returned iterator poitn to last valid element. Then you don't need to create `cleaned_str` just you will copy form `str` to `str` :)